src/sdl2/render.rs

```diff
@@ -164,14 +166,43 @@ impl TryFrom<u32> for BlendMode {
         use self::BlendMode::*;
         use crate::sys::SDL_BlendMode::*;
 
-        Ok(match unsafe { transmute(n) } {
-            SDL_BLENDMODE_NONE => None,
-            SDL_BLENDMODE_BLEND => Blend,
-            SDL_BLENDMODE_ADD => Add,
-            SDL_BLENDMODE_MOD => Mod,
-            SDL_BLENDMODE_MUL => Mul,
-            SDL_BLENDMODE_INVALID => Invalid,
-        })
+        match n {
+            x if x == SDL_BLENDMODE_NONE as u32 => Ok(None),
+            x if x == SDL_BLENDMODE_BLEND as u32 => Ok(Blend),
+            x if x == SDL_BLENDMODE_ADD as u32 => Ok(Add),
+            x if x == SDL_BLENDMODE_MOD as u32 => Ok(Mod),
+            x if x == SDL_BLENDMODE_MUL as u32 => Ok(Mul),
+            x if x == SDL_BLENDMODE_INVALID as u32 => Ok(Invalid),
+            _ => Err(()),
+        }
+    }
+}
+
+#[repr(i32)]
```

Comment:
This can be left out, the compiler will then choose to only make it 1 byte wide instead of 4.

Reply:
I'm following the standard in this repo. Everywhere else does it like this, and it's not up to me to make a sweeping.

Reply:
it deduces the type as an isize instead if it's not stated, not a byte. Prefer to keep it like this for consistency

Reply:
> it deduces the type as an isize instead if it's not stated

Enum values are given as `isize` values, the compiler will then find out everything fits in `u8`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d25ede470171a863fa03dad25e7c3d6a

Reply:
Huh, neat. TIL 7db255048a756518c52d5b78619df634b7eb3487